src/lib.rs

```diff
@@ -692,6 +680,45 @@ impl Client {
         Ok(client)
     }
 
+    /// Create a new client with additional headers
+    /// # Errors
+    pub fn new_with_headers(base_url: &str, additional_headers: HeaderMap) -> Result<Self, Error> {
+        let mut client = Self::new(base_url)?;
+        let mut headers = Self::default_http_headers();
+
+        for (key, value) in additional_headers {
+            match key {
+                None => return Err(Error::InvalidResponse),
+                Some(k) => headers.insert(k, value),
+            };
```

Comment:
See: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or
```suggestion
            headers.insert(key.ok_or(Error::InvalidResponse)?, value);
```